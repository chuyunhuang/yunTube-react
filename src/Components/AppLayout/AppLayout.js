import React from 'react';
import './AppLayout.scss';
import HeaderNav from '../../Containers/HeaderNav/HeaderNav';


export function AppLayout(props) {
  return (
    <div className="app-layout">
      <HeaderNav />
      {props.children}
    </div>
  )
}