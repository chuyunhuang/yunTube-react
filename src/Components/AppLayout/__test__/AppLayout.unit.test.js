import React from 'react';
import { shallow } from 'enzyme';
import { AppLayout } from '../AppLayout';

describe('AppLayout', () => {
  test('renders <AppLayout /> ', () => {
    const wrapper = shallow(
      <AppLayout />
    )
    expect(wrapper).toMatchSnapshot();
  });

  test('renders <AppLayout /> with one child', () => {
    const wrapper = shallow(
      <AppLayout >
        <div>Child 1</div>
      </AppLayout >
    )
    expect(wrapper).toMatchSnapshot();
  });

  test('renders <AppLayout /> with more than one children', () => {
    const wrapper = shallow(
      <AppLayout>
        <div>Child 1</div>
        <div>
          <span>Child 2</span>
          <div>Child 3</div>
        </div>
      </AppLayout>
    )
    expect(wrapper).toMatchSnapshot();
  });
});