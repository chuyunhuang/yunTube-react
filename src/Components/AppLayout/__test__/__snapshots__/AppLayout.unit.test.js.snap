// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppLayout renders <AppLayout />  1`] = `
<div
  className="app-layout"
>
  <HeaderNav />
</div>
`;

exports[`AppLayout renders <AppLayout /> with more than one children 1`] = `
<div
  className="app-layout"
>
  <HeaderNav />
  <div>
    Child 1
  </div>
  <div>
    <span>
      Child 2
    </span>
    <div>
      Child 3
    </div>
  </div>
</div>
`;

exports[`AppLayout renders <AppLayout /> with one child 1`] = `
<div
  className="app-layout"
>
  <HeaderNav />
  <div>
    Child 1
  </div>
</div>
`;
